libcudacxx/include/cuda/std/__mdspan/layout_stride.h

```diff
@@ -331,17 +327,13 @@ struct layout_stride
        */
     }
 
-    __MDSPAN_TEMPLATE_REQUIRES(
+    // nvcc cannot deduce this constructor when using _LIBCUDACXX_REQUIRES
+    template <
       class _IntegralTypes,
-      /* requires */ (
-        // MSVC 19.32 does not like using index_type here, requires the typename _Extents::index_type
-        // error C2641: cannot deduce template arguments for '_CUDA_VSTD::layout_stride::mapping'
-        _CCCL_TRAIT(_CUDA_VSTD::is_convertible, const remove_const_t<_IntegralTypes>&, typename _Extents::index_type)
-        && _CCCL_TRAIT(
-          _CUDA_VSTD::is_nothrow_constructible, typename _Extents::index_type, const remove_const_t<_IntegralTypes>&)))
-    __MDSPAN_INLINE_FUNCTION
-    constexpr mapping(extents_type const& __e,
-                      _CUDA_VSTD::span<_IntegralTypes, extents_type::rank()> const& __s) noexcept
+      enable_if_t<_CCCL_TRAIT(is_convertible, const remove_const_t<_IntegralTypes>&, index_type), int>           = 0,
+      enable_if_t<_CCCL_TRAIT(is_nothrow_constructible, index_type, const remove_const_t<_IntegralTypes>&), int> = 0>
```

Comment:
This might have been the breaking thing, because nvcc has issues deducing the template arguments here otherwise